app/components/post_list/post_list.tsx

```diff
@@ -109,6 +140,8 @@ const PostList = ({
     const scrolledToHighlighted = useRef(false);
     const [enableRefreshControl, setEnableRefreshControl] = useState(false);
     const [refreshing, setRefreshing] = useState(false);
+    const [showScrollToEndBtn, setShowScrollToEndBtn] = useState(false);
+    const [isNewMessages, setIsNewMessages] = useState(false);
```

Comment:
In my opinion there is no need to have to re-render one of the most heavy components in the app just for this, but in order to be able to display the component either with the down chevron or the new messages text, you can extract it to it's own component instead and use references to update it